backend/core/src/listings/listings.controller.ts

```diff
@@ -60,14 +65,9 @@ export class ListingsController {
 
   @Get()
   @ApiOperation({ summary: "List listings", operationId: "list" })
-  @ApiImplicitQuery({
-    name: "jsonpath",
-    required: false,
-    type: String,
-  })
   @UseInterceptors(CacheInterceptor)
-  async list(@Query() queryParams: ListingsListQueryParams): Promise<ListingDto[]> {
-    return mapTo(ListingDto, await this.listingsService.list(queryParams))
+  async list(@Query() queryParams: ListingsListQueryParams): Promise<PaginatedListingsDto> {
+    return await this.listingsService.list(queryParams)
```

Comment:
Do you ever use the PaginatedListings object? Could you just unwrap it here and not have to change the API so it still returns Promise<ListingDto[]>?

Reply:
We don't yet use the PaginatedListingsDto, but we will want to in the /listings frontend: we'll want to display what page we're on, whether there is a next page (or we're on the last page), maybe how many total listings there are across all pages.

Reply:
Will this also force us to change any implementation in the partner portal that fetches listings? E.g will we have to incorporate pagination into the partners' page to view their own listings?

Reply:
Yeah good call, we will have to update the parts of the partner portal that call `GET /listings` (similar to the one-line changes I made to the `/sites/public/pages` files in this PR). But if we don't want to do the frontend work to the partner portal to add pagination, we can just explicitly set the params on the `GET /listings` requests to e.g. `page: -1, limit: -1` and that will make it so that the results include all listings.

TLDR: we'll need to make a couple one-line changes to merge Exygy's partner-portal-listings work into our repo.

Reply:
This is probably a big ask, but it may be worth it to decouple the queries so we have the ability to request both paginated and unpaginated versions of listings. WDYT?

Reply:
Or you could make the defaults `page: -1, limit: -1` such that `GET /listings` will return the same as it would today?

Reply:
@willrlin that seems reasonable, given that AFAICT pagination is currently only used for applications, which we're not using anyway.

Reply:
Thanks for these suggestions! Just pushed a commit that makes the default values `page: -1, limit: -1`. I had to make changes to the ApplicationService so that the negative values wouldn't break it. With those changes, both `yarn test` and `yarn test:e2e:local` pass.

Reply:
Hm, so does that mean that the partner portal frontend is not calling with page/limit because they're using the default? Maybe we should file a task to consider putting 10 as the default limit in `ApplicationService` (i.e. `limit > 0 ? limit : 10`) instead of max integer.

Reply:
Updated to make the default in ApplicationService 10 instead of max integer.